src/Cli/dotnet/ParseResultExtensions.cs

```diff
@@ -64,15 +64,35 @@ public static string[] GetSubArguments(this string[] args)
             var runArgs = subargs.Contains("--") ? subargs.GetRange(subargs.IndexOf("--"), subargs.Count() - subargs.IndexOf("--")) : new List<string>();
             subargs = subargs.Contains("--") ? subargs.GetRange(0, subargs.IndexOf("--")) : subargs;
 
-            subargs.RemoveAll(arg => DiagOption.Aliases.Contains(arg));
-            if (subargs[0].Equals("dotnet"))
-            {
-                subargs.RemoveAt(0);
-            }
+            subargs = subargs.SkipWhile(arg => DiagOption.Aliases.Contains(arg) || arg.Equals("dotnet")).ToList();
             subargs.RemoveAt(0); // remove top level command (ex build or publish)
             return subargs.Concat(runArgs).ToArray();
         }
 
+        public static bool DiagOptionPrecedesSubcommand(this string[] args, string subCommand)
```

Comment:
Not suggesting you write the method this way, but when I see a bunch of boolean logic returning hard-coded blooeans, I always try to find a way to simplify. Here, I used `Any` for short circuiting and use a variable to flip the logic for the truthy result:
```C#
public static bool DiagOptionPrecedesSubcommand(this string[] args, string subCommand)
{
    if (string.IsNullOrEmpty(subCommand))
    {
        return true;
    }

    var isAlias = false;
    return args.Any(arg => arg.Equals(subCommand) || (isAlias = DiagOption.Aliases.Contains(arg))) && isAlias;
}
```
Ends up with the same results where if the `Equals` is true, the result is false because `true && false` is false. If `Contains` is true, sets `isAlias` to true and thus `true && true` is true.

Reply:
This is definitely clever, and I love how short it is! Some people really don't like setting a variable inside a condition, and I'd admit in this case I had to read it a few times to understand why what you wrote really is both correct and equivalent. (The first time I read it, I thought you'd missed that arg.Equals(subCommand) should return **false** instead of **true**...though I'll admit I read the code before reading your last sentence.)

I'd been thinking about cutting out the for loop with a couple IndexOf-type statements, but I think this is the clearest version of the three. (I can remove the redundant elses, though.)

Reply:
![image](https://media.tenor.com/GDN1CUsgRigAAAAC/the-more.gif)